sopel/web.py

```diff
@@ -237,3 +237,43 @@ def iri_to_uri(iri):
     urlencode = urllib.urlencode
 else:
     urlencode = urllib.parse.urlencode
+
+
+def trim_url(url):
+    # clean trailing sentence- or clause-ending punctuation
+    while url[-1] in '.,?!\'":;':
+        url = url[:-1]
+
+    # clean unmatched parentheses/braces/brackets
+    for (opener, closer) in [('(', ')'), ('[', ']'), ('{', '}'), ('<', '>')]:
+        if url[-1] == closer and url.count(opener) < url.count(closer):
+            url = url[:-1]
+
+    return url
+
+
+def search_urls(text, exclusion_char=None, clean=False, schemes=None):
```

Comment:
I've this feeling that the function name should be `parse_urls` or `find_urls` instead… Thoughts?

Reply:
I'd argue that `parse_urls` should parse URLs, ie. get a list of URLs and extract something from them. `find_urls` on the other hand would do the same thing as `search_urls`, but I can't see the difference. Would you tell me what your gut feeling are saying?